core/src/server/access_control/host.rs

```diff
@@ -188,43 +173,34 @@ mod tests {
 		assert_eq!(Host::parse("chrome-extension://124.0.0.1"), Host::new("124.0.0.1", None));
 		assert_eq!(Host::parse("parity.io/somepath"), Host::new("parity.io", None));
 		assert_eq!(Host::parse("127.0.0.1:8545/somepath"), Host::new("127.0.0.1", Some(8545)));
-	}
 
-	#[test]
-	fn should_reject_when_there_is_no_header() {
-		let valid = is_host_valid(None, &AllowHosts::Any);
-		assert!(!valid);
-		let valid = is_host_valid(None, &AllowHosts::Only(vec![]));
-		assert!(!valid);
+		let host = Host::parse("*.domain:*/somepath");
+		assert_eq!(host.port, Port::Pattern("*".into()));
+		assert_eq!(host.hostname.as_str(), "*.domain");
 	}
 
 	#[test]
-	fn should_reject_when_validation_is_disabled() {
-		let valid = is_host_valid(Some("any"), &AllowHosts::Any);
-		assert!(valid);
+	fn should_allow_when_validation_is_disabled() {
+		assert!((AllowHosts::Any).verify("any").is_ok());
 	}
 
 	#[test]
 	fn should_reject_if_header_not_on_the_list() {
-		let valid = is_host_valid(Some("parity.io"), &AllowHosts::Only(vec![]));
-		assert!(!valid);
+		assert!((AllowHosts::Only(vec![].into())).verify("parity.io").is_err());
 	}
 
 	#[test]
 	fn should_accept_if_on_the_list() {
-		let valid = is_host_valid(Some("parity.io"), &AllowHosts::Only(vec!["parity.io".into()]));
-		assert!(valid);
+		assert!((AllowHosts::Only(vec!["parity.io".into()].into())).verify("parity.io").is_ok());
```

Comment:
if it's on the list without a port, like this, would `.verify("parity.io:1234")` work? In other words, is any port valid if no port is given in the allow list?

Reply:
no it will be denied unless the port is whitelisted or `*` is used but I will add a test to make it clear.